spec/requests/admin/commodities_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+RSpec.describe Api::Admin::CommoditiesController do
+  describe 'GET #index' do
+    subject(:do_request) do
+      get api_commodities_path(format: :csv)
+      response.body
+    end
+
+    before do
+      allow(TimeMachine).to receive(:at).and_call_original
+    end
+
+    it 'contains the correct headers' do
+      do_request
+
+      expect(response.body).to include("SID,Commodity code,Product line suffix,Description,Start date,End date,Indentation,End line,ItemIDPlusPLS\n")
+    end
+
+    it 'calls the TimeMachine with the correct Date' do
```

Comment:
This actually doesn't add value (if you remember from our conversation previously). You're also using a different mechanism for filtering based on the actual date: you pass the actual date as a parameter to the query

Reply:
I'd validate that our service receives the actual date

Reply:
```ruby
it 'passes the actual date to the query service' do
  allow(Admin::QueryAllCommodities).to receive(:call)
  do_request
  expect(Admin::QueryAllCommodities).to receive(:call).with(expected_actual_date)
end
```

Reply:
Yes, I remember now.
thanks for writing the test!
( I've just changed `to receive` -> `to have_received` )